rootfs-builder/rootfs.sh

```diff
@@ -307,17 +327,46 @@ build_rootfs_distro()
 
 	mkdir -p ${ROOTFS_DIR}
 
+	# need to detect rustc's version too?
 	detect_go_version ||
 		die "Could not detect the required Go version for AGENT_VERSION='${AGENT_VERSION:-master}'."
 
 	echo "Required Go version: $GO_VERSION"
 
+	detect_rust_version ||
+		die "Could not detect the required rust version for AGENT_VERSION='${AGENT_VERSION:-master}'."
+
+	echo "Required rust version: $RUST_VERSION"
+
+	detect_cmake_version ||
+		die "Could not detect the required cmake version for AGENT_VERSION='${AGENT_VERSION:-master}'."
+
+	echo "Required cmake version: $CMAKE_VERSION"
+
+	detect_musl_version ||
+		die "Could not detect the required musl version for AGENT_VERSION='${AGENT_VERSION:-master}'."
+
+	echo "Required musl version: $MUSL_VERSION"
+
 	if [ -z "${USE_DOCKER}" ] && [ -z "${USE_PODMAN}" ]; then
 		#Generate an error if the local Go version is too old
 		foundVersion=$(go version | sed -E "s/^.+([0-9]+\.[0-9]+\.[0-9]+).*$/\1/g")
 
 		compare_versions "$GO_VERSION" $foundVersion || \
 			die "Your Go version $foundVersion is older than the minimum expected Go version $GO_VERSION"
+
+		if [ "${RUST_AGENT}" == "yes" ]; then
+			source /root/.cargo/env
+			foundVersion=$(rustc --version | sed -E "s/^.+([0-9]+\.[0-9]+\.[0-9]+).*$/\1/g")
+
+			compare_versions "$RUST_VERSION" $foundVersion || \
```

Comment:
if you are `""`ing `RUST_VERSION`, you should probably quote `foundVersion` as well, for symmetry if nothing else :-)

Reply:
Let me check, I recall I copied this from somewhere

Reply:
Yes, I'd quote all variables to avoid nasty surprises :)